### PR TITLE
Flp 558 authorisation in data layer

### PR DESCRIPTION
**Pull request by dinathain**
Added authorisation to ensure that bearer tokens in API calls are validated, in addition to being used to determine which data can be queried/updated

### CODE REVIEW

File: src/InnerApi/Identity/Authorization/BearerTokenMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            await _next(context);
+        }
+
+        private bool DisableAccountAuthorisation()
+        {
+            return bool.TryParse(_configuration["DisableAccountAuthorisation"], out var disableAccountAuthorisation) && disableAccountAuthorisation;
```
Comment (tom-gough) [inline src/InnerApi/Identity/Authorization/BearerTokenMiddleware.cs:44]: we seem to be using a combination of american and british english for authorisation/authorization, worth standardising? I think elsewhere we are using z to match the c# standard

**Review by tom-gough — APPROVED** [review]

File: src/Domain/Repositories/ApprenticeshipQueryRepository.cs [hunk 2]
```diff
@@ -63,19 +62,9 @@ public async Task<IEnumerable<DataTransferObjects.Apprenticeship>> GetAll(long u
...
-                .Select(PriceHistoryToApprenticeshipPrice())
-                .ToListAsync();
-
-            return dataModels;
-        }
-
         public async Task<PendingPriceChange?> GetPendingPriceChange(Guid apprenticeshipKey)
         {
-            _logger.LogInformation($"Getting pending price change for apprenticeship {apprenticeshipKey}");
+            _logger.LogInformation("Getting pending price change for apprenticeship {p1}", apprenticeshipKey);
```
Comment (MikeV-TalentConsulting) [inline src/Domain/Repositories/ApprenticeshipQueryRepository.cs:67]: Why are we labelling apprenticeshipKey p1?

File: src/Infrastructure/IAccountIdClaimsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using SFA.DAS.Apprenticeships.Enums;
+
+namespace SFA.DAS.Apprenticeships.Infrastructure
```
Comment (MikeV-TalentConsulting) [inline src/Infrastructure/IAccountIdClaimsHandler.cs:3]: Optional, could remove curly braces on namespaces for any new classes

File: src/InnerApi/Controllers/ApprenticeshipController.cs [hunk 4]
```diff
@@ -101,7 +103,11 @@ public async Task<IActionResult> GetApprenticeshipKeyByApprenticeshipId(long app
         {
 	        var request = new GetApprenticeshipKeyByApprenticeshipIdRequest { ApprenticeshipId = apprenticeshipId };
 	        var response = await _queryDispatcher.Send<GetApprenticeshipKeyByApprenticeshipIdRequest, GetApprenticeshipKeyByApprenticeshipIdResponse>(request);
-	        if (response.ApprenticeshipKey == null) return NotFound();
+            if (response.ApprenticeshipKey == null)
+            {
+                _logger.LogInformation("{p1} could not be found.", nameof(response.ApprenticeshipKey));
```
Comment (MikeV-TalentConsulting) [inline src/InnerApi/Controllers/ApprenticeshipController.cs:108]: Same as other comment

**Review by MikeV-TalentConsulting — APPROVED** [review]

**Pull request merged**